### PR TITLE
Refactors job EXP to allow for multiple departments in restrictions

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Refactors job EXP to allow for multiple departments to be included in the restrictions process (IE: 50 hours in medical AND 20 hours in security).

Spawned from a suggestions thread on the forums which may allow us to more tightly refine was requirements are needed on jobs. If you have multiple departments waiting, it will show restrictions for those you have not gotten the required time in yet. If you only lack the time in one department, only that one will be shown.
image
image
image

This PR also removes the exempt EXP type. Only 2 people have it, and both of them have long surpassed the hours needed to do things. On top of that, one is an admin, and the other is a retired admin who hasnt logged in in over a year. 

# IMPORTANT

This PR does **NOT** alter the existing time limits, it merely adds framework for expanding upon it in the future. This is so that this PR can stay inside refactor territory. 

## Why It's Good For The Game
Having job restrictions be bound to only one department is kind of limiting for jobs which should require experience in multiple departments. Even if we dont need that now, I like to think in the long run since we will need more jobs soon.

## Changelog
N/A, refactor

### CODE REVIEW

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -253,14 +253,20 @@
...
-	if(!is_job_whitelisted(src, rank))	 return 0
-	if(!job.player_old_enough(client))	return 0
-	if(job.admin_only && !(check_rights(R_EVENT, 0))) return 0
-	if(job.available_in_playtime(client))
-		return 0
+	if(!job)
+		return FALSE
+	if(!job.is_position_available())
+		return FALSE
+	if(jobban_isbanned(src,rank))
```
Comment (dearmochi) [inline code/modules/mob/new_player/new_player.dm:260]:
```suggestion
	if(jobban_isbanned(src, rank))
```

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -253,14 +253,20 @@
...
+		return FALSE
+	if(!is_job_whitelisted(src, rank))
+		return FALSE
+	if(!job.player_old_enough(client))
+		return FALSE
+	if(job.admin_only && !(check_rights(R_EVENT, FALSE)))
```
Comment (dearmochi) [inline code/modules/mob/new_player/new_player.dm:266]:
```suggestion
	if(job.admin_only && !check_rights(R_EVENT, FALSE))
```

File: code/game/jobs/job_exp.dm [hunk 3]
```diff
@@ -122,31 +119,52 @@ GLOBAL_LIST_INIT(role_playtime_requirements, list(
 	return max(0, req_mins - my_exp)
 
 
-/datum/job/proc/available_in_playtime(client/C)
+/datum/job/proc/isPlayable(client/C)
```
Comment (dearmochi) [inline code/game/jobs/job_exp.dm:122]: No more `snake_case` proc name? Why?

Comment (dearmochi) [inline code/game/jobs/job_exp.dm:122]: No more `snake_case` proc name? Why?

File: code/game/jobs/job_exp.dm [hunk 4]
```diff
@@ -122,31 +119,52 @@ GLOBAL_LIST_INIT(role_playtime_requirements, list(
...
+	for(var/exp_type in exp_map)
+		if(!(exp_type in play_records))
+			success = FALSE
+			continue
+		if(text2num(exp_map[exp_type]) > text2num(play_records[exp_type]))
+			success = FALSE
+
+	return success
+
+/datum/job/proc/getRestrictions(client/C)
```
Comment (dearmochi) [inline code/game/jobs/job_exp.dm:146]: Ditto, plus the name is rather vague.

**Review by dearmochi — CHANGES_REQUESTED** [review]

File: code/game/jobs/job_exp.dm (hunk 3 above)
Comment (AffectedArc07) [reply code/game/jobs/job_exp.dm:122]: This became my default casing after doing so much java

Comment (AffectedArc07) [reply code/game/jobs/job_exp.dm:122]: Ditto

File: code/game/jobs/job_exp.dm (hunk 4 above)
Comment (AffectedArc07) [reply code/game/jobs/job_exp.dm:146]: Ill make it `get_exp_restrictions`

**Review by dearmochi — APPROVED** [review]

**Pull request merged**